src/en/mangaowlto/src/eu/kanade/tachiyomi/extension/en/mangaowlto/MangaOwlToFactory.kt

```diff
@@ -0,0 +1,105 @@
+package eu.kanade.tachiyomi.extension.en.mangaowlto
+
+import eu.kanade.tachiyomi.source.Source
+import eu.kanade.tachiyomi.source.SourceFactory
+import eu.kanade.tachiyomi.source.model.Filter
+
+class MangaOwlToFactory : SourceFactory {
+    class Genre(name: String, val uriPart: String) : Filter.CheckBox(name)
```

Comment:
Genre data should not be `Filter` classes, otherwise the filter state can't be reset. Also move this out of the factory. Move your filter classes (including the ones in `MangaOwlTo.kt`) into a separate file (`Filters.kt` or `MangaOwlToFilters.kt`).

Reply:
This should allow reseting the filter.
> class GenresFilter(genres: List<Genre>) :
>    Filter.Group<Genre>("Genres", genres.map { Genre(it.name, it.uriPart) })
Almost other extensions are also use Genre as Filter.CheckBox so I don't know if any other method is possible.

Reply:
Use this to store data
```kotlin
class Genre(val name: String, val uriPart: String)
```
and map to this
```kotlin
class GenreCheckBox(name: String, val uriPart: String) : Filter.CheckBox(name)
```

Don't reuse the same class for multiple purposes.